velociraptor/observations/objects.py

```diff
@@ -569,13 +569,16 @@ def plot_on_axes(self, axes: Axes, errorbar_kwargs: Union[dict, None] = None):
         elif self.plot_as == "line":
             kwargs["zorder"] = line_zorder
 
+        # Make both the data name and redshift appear in the legend
+        data_label = self.citation + " ($z={:.1f}$)".format(self.redshift)
```

Comment:
Please avoid "adding" strings to concatenate them. This should be:

`data_label = f"{self.citation} ($z={self.redshift:.1f}$)"`

Reply:
Done